### PR TITLE
🧹 Update documentation

### PR DESCRIPTION
**Pull request by pje**
- Document the release process in `CONTRIBUTING.md`
- Add development instructions to `CONTRIBUTING.md` (adapted from [download-artifact](https://github.com/actions/download-artifact))
- Add a test badge to the `README`
- Link to `CONTRIBUTING.md` in `README`
- Unify copy between `README.md` and `action.yml`
- Copy-pasta the CoC from [download-artifact](https://github.com/actions/download-artifact)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,26 +1,69 @@
...
+
+1. Bump the package version by running [`npm version [major|minor|patch]`](https://docs.npmjs.com/cli/v7/commands/npm-version). We adhere to [SemVer 2.0](https://semver.org/spec/v2.0.0.html) to the best of our ability. Commit the changes to `package.json` and `package-lock.json` and push them to main.
+1. [Draft a new release](https://github.com/actions/labeler/releases/new) pointing to the ref of the version bump you just made. Publish the release to the marketplace when complete.
+1. Finally: update the most recent "shorthand tag" (v1, v2, v3, etc) to point to the specific ref of the release you just made. For example, if we just released `v1.1.0`, we would rewrite the `v1` tag like this:
+
+```
+git tag -fa v1 v1.1.0 -m "Update v1 tag to point to v1.1.0"
+git push origin v1 --force
+```
+
```
Comment (pje) [inline CONTRIBUTING.md:55]: @TingluoHuang How does this sound to you?

Comment (TingluoHuang) [reply CONTRIBUTING.md:55]: I would like to get @thboop to take a look since I haven't done lots of action releases.

**Comment by pje** [comment]
https://github.com/github/c2c-actions-runtime/issues/1273

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,26 +1,69 @@
...
+## Releasing a new version
+
+All the concepts from [the actions/toolkit release docs](https://github.com/actions/toolkit/blob/main/docs/action-versioning.md) apply. Please read that first!
+
+Once the changes are merged into main, a repo maintainer should:
+
+1. Bump the package version by running [`npm version [major|minor|patch]`](https://docs.npmjs.com/cli/v7/commands/npm-version). We adhere to [SemVer 2.0](https://semver.org/spec/v2.0.0.html) to the best of our ability. Commit the changes to `package.json` and `package-lock.json` and push them to main.
+1. [Draft a new release](https://github.com/actions/labeler/releases/new) pointing to the ref of the version bump you just made. Publish the release to the marketplace when complete.
+1. Finally: update the most recent "shorthand tag" (v1, v2, v3, etc) to point to the specific ref of the release you just made. For example, if we just released `v1.1.0`, we would rewrite the `v1` tag like this:
```
Comment (thboop) [inline CONTRIBUTING.md:49]:
```suggestion
1. Finally: update the corresponding "major tag" (v1, v2, v3, etc) to point to the specific ref of the release you just made. For example, if we just released `v1.1.0`, we would rewrite the `v1` tag like this:
```

Pretty nit: In most scenarios the most recent would be fine, but these actually correspond to major versions, and occasionally we may do a backport fix, so if you backported a fix to `v1.6` when `v2.0` is already out, you would want to update `v1`

**Review by thboop — APPROVED** [review]
LGTM with one minor Nit

File: CONTRIBUTING.md (hunk 2 above)
Comment (pje) [reply CONTRIBUTING.md]: Great point, thanks @thboop!

**Review by sekeroz — COMMENTED** [review]
Haini

**Review by sekeroz — COMMENTED** [review]
##  ****

**Pull request merged**